From d2ec4940325f3942cff88ee769e39a0345c65418 Mon Sep 17 00:00:00 2001
From: Logan Chien <loganchien@google.com>
Date: Thu, 11 Oct 2012 17:00:58 +0800
Subject: [PATCH] Enable PIC for Android even if "-static" is specified.

Android only supports a limited subset of the relocation
types on ARM and MIPS architecture.  Without this fix,
the Android build system can't link the executable properly.
---
 lib/Driver/Tools.cpp |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/lib/Driver/Tools.cpp b/lib/Driver/Tools.cpp
index 11df5dd..8cee28f 100644
--- a/lib/Driver/Tools.cpp
+++ b/lib/Driver/Tools.cpp
@@ -1585,7 +1585,7 @@ void Clang::ConstructJob(Compilation &C, const JobAction &JA,
   // PIC or PIE options above, if these show up, PIC is disabled.
   if (Args.hasArg(options::OPT_mkernel))
     PICDisabled = true;
-  if (Args.hasArg(options::OPT_static))
+  if (Args.hasArg(options::OPT_static) && !isAndroid)
     PICDisabled = true;
   bool DynamicNoPIC = Args.hasArg(options::OPT_mdynamic_no_pic);
 
-- 
1.7.7.3

